src/components/ListView.tsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+    Dispatch,
+    SetStateAction,
+    createContext,
+    memo,
+    useCallback,
+    useContext,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { useKeyPress, useMounted } from '@taskany/bricks';
+
+interface ListViewProps {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onClick?: (value: any) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onKeyboardClick?: (value: any) => void;
+    children: React.ReactNode;
+}
+
+interface ListViewContext {
+    registerItem: (value: unknown) => number;
+    items: unknown[];
+    cursor?: number;
+    hovered?: number;
+    setHovered?: Dispatch<SetStateAction<undefined | number>>;
+}
+const listViewContext = createContext<ListViewContext>({
+    registerItem: () => 0,
+    items: [],
+});
+
+export const ListView: React.FC<ListViewProps> = ({ children, onKeyboardClick }) => {
+    const downPress = useKeyPress('ArrowDown');
+    const upPress = useKeyPress('ArrowUp');
+    const spacePress = useKeyPress(' ');
+    const enterPress = useKeyPress('Enter');
+    const [cursor, setCursor] = useState<number | undefined>();
+    const [hovered, setHovered] = useState<number | undefined>();
+    const items: unknown[] = useMemo(() => [], []);
+
+    const registerItem = useCallback((value: unknown) => items.push(value) - 1, [items]);
```

Comment:
why you are changing memoized variable instead of `useState` returned value through setter function?

Reply:
or use returned value from `useRef` function

Reply:
also this code is not safety for multiple re-renders of the `ListViewItem` component